Search Bar to fetch tracks from iTunes

mashrukh-islam opened the pull request:
This search bar fetches tracks from the iTunes API based on the user's search query.
A new container called ITunesSearch is created using flocki. This container contains
a search bar from antd.

### Ticket Link
---------------------------------------------------


### Related Links
---------------------------------------------------


### Description
---------------------------------------------------


### Steps to Reproduce / Test
---------------------------------------------------


### GIF's
---------------------------------------------------
![SearchBar](https://user-images.githubusercontent.com/87057006/124932327-9c40cd80-e020-11eb-85e7-208e531383d5.gif)

Commit:
Search Bar to fetch tracks from iTunes

This search bar fetches tracks from the iTunes API based on the user's search query.
A new container called ITunesSearch is created using flocki. This container contains
a search bar from antd.

alichherawalla commented on app/containers/App/index.js:
```diff
@@ -31,27 +28,7 @@ export function App({ location }) {
     <ThemeProvider theme={theme}>
       <Header />
       <Layout.Content>
-        <For
```
lets continue using the routing logic

alichherawalla commented on app/containers/ITunesSearch/index.js:
```diff
@@ -0,0 +1,132 @@
+  dispatchFetchTracks,
+  dispatchClearTracks,
+  artistName,
+  tracks = {},
```
this is an issue in the react-template as well, use the default props instead of assigning a default value here. 

if you can create a PR fixing it in the react-template too that would be awesome!

mashrukh-islam committed:
Routing logic and default props for container

Brought back the routing logic in the App container.Set default props
of ITunesSearch container.

alichherawalla commented on app/containers/ITunesSearch/index.js:
```diff
@@ -0,0 +1,127 @@
+  const renderTracks = () => {
+    const trackList = get(tracks, 'results', []);
+    return (
+      (trackList.length !== 0 || loading) && (
```
(trackList.length || loading)

alichherawalla commented on app/containers/ITunesSearch/index.js:
```diff
@@ -0,0 +1,127 @@
+    return (
+      (trackList.length !== 0 || loading) && (
+        <Skeleton loading={loading} active>
+          {trackList.map((item, index) => (
```
`For`

mashrukh-islam committed:
Fixed if condition and used For component

Fixed if condition while rendering results from the search bar.
Used For component to render results from user's search query.
